qiskit/providers/ibmq/managed/managedjobset.py

```diff
@@ -0,0 +1,287 @@
+            backend: Backend to execute the experiments on.
+            executor: The thread pool to use.
+            assemble_config: Additional arguments used to configure the Qobj
+                assembly. Refer to the ``qiskit.compiler.assemble`` documentation
+                for details on these arguments.
+
+        Raises:
+            IBMQJobManagerInvalidStateError: If the jobs were already submitted.
+        """
+        if self._managed_jobs:
+            raise IBMQJobManagerInvalidStateError("Jobs were already submitted.")
+
+        exp_index = 0
+        for i, experiment in enumerate(experiment_list):
+            qobj = assemble(experiment, backend=backend, **assemble_config)
+            job_name = "{}_{}_".format(self._name, i)
+            future = executor.submit(
+                self._async_submit, qobj=qobj, job_name=job_name, backend=backend)
+            self._managed_jobs.append(
+                ManagedJob(experiment, start_index=exp_index, future=future))
+            exp_index += len(experiment)
+
+    def _async_submit(
+            self,
+            qobj: Qobj,
+            job_name: str,
+            backend: IBMQBackend,
+    ) -> IBMQJob:
+        """Run a Qobj asynchronously.
+
+        Args:
+            qobj: Qobj to run.
+            job_name: Name of the job.
+            backend: Backend to execute the experiments on.
+
+        Returns:
+            IBMQJob instance for the job.
+        """
+        return backend.run(qobj=qobj, job_name=job_name)
+
+    def submit_results(self) -> None:
+        """Collect job submit responses.
+
+        Raises:
+            IBMQJobManagerInvalidStateError: If jobs have not been submitted.
+        """
+        if not self._managed_jobs:
+            raise IBMQJobManagerInvalidStateError("Jobs need to be submitted first!")
+
+        if all(mjob.future is None for mjob in self._managed_jobs):
+            # Submit results already collected
+            return
+
+        for managed in self._managed_jobs:
+            managed.submit_result()
+
+    @requires_submit
+    def statuses(self) -> List[Union[JobStatus, None]]:
+        """Return the status of each job.
+
+        Returns:
+            A list of job statuses. The entry is ``None`` if the job status
+                cannot be retrieved due to server error.
+        """
+        statuses = []
+        for i, mjob in enumerate(self._managed_jobs):
+            if mjob.job is not None:
+                try:
+                    statuses.append(mjob.job.status())
+                    continue
+                except JobError as err:
+                    warnings.warn(
+                        "Unable to retrieve status for job {}, job ID={}, "
+                        "for experiments {}-{}: {}".format(
+                            i, mjob.job.job_id(), mjob.start_index, mjob.end_index, err))
+            statuses.append(None)
+
+        return statuses
+
+    @requires_submit
```

Comment:
This `requires_submit` would make `report()` not really return immediately, right? Thinking that part of the usefulness of the manager will come from submitting a large number of jobs, I think we should try to just add a summary for "jobs that have been created but are waiting for the executor to have a free spot" - I don't mind squeezing it into `Pending jobs:` or have a new line entirely.

Reply:
If `report()` and `statuses()` don't wait for submit results, then a separate thread needs to wait and collect them. Since the users are likely to invoke `report()` or `statuses()` to check the progress of their jobs.